ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -751,7 +751,7 @@ static char *make_filename_safe(const char *filename)
 		}
 		return estrdup(filename);
 	}
-	if (*filename && memcmp(filename, ":memory:", sizeof(":memory:"))) {
+	if (*filename && zend_binary_strcmp(filename, strlen(filename)-1, ":memory:", strlen(":memory:")-1)) {
```

Comment:
Subtracting 1 from strlen is not necessary, the subtraction of 1 from sizeof is because sizeof also counts the NULL terminator but strlen does not.

Reply:
I was thinking about that same kind of confusion, just wanted to confirm.

Reply:
It seems like I'm a little too sleepy....Sorry for making me check again and again. (AM 1:43 JST)

Reply:
it s ok Saki :) relax, as nielsdos mentioned, an "attacker" can t do anything really malicious here. no rush :)